### PR TITLE
feat: support keycloak login to wagtail admin

### PR DESCRIPTION
**Pull request by olemathias**
As we move towards production not having to deal with local users and passwords is prefered.
This pull request implements authentication towards Wannabe5.

Will also work for django-admin but the login templates is not modified, so will first have to login to wagtail admin and then navigate to django-admin.

No permissions are granted on first login, so a admin will have to grant permissions after the user is created.

With local login disabled and wannabe login enabled:
bilde

With local and wannabe login enabled:
bilde

### CODE REVIEW

File: tgno/settings/base.py [hunk 1]
```diff
@@ -176,6 +184,26 @@
...
+
+# Custom settings
+SITE_NAME = env("SITE_NAME", "TG.no")
+DISABLE_LOCAL_AUTH = env("DISABLE_LOCAL_AUTH", "False").lower() in ("true", "1")
+
+# Django Social Auth
+AUTHENTICATION_BACKENDS = []
+
+SOCIAL_AUTH_JSONFIELD_ENABLED = True
+SOCIAL_AUTH_KEYCLOAK_KEY = os.getenv("SOCIAL_AUTH_KEYCLOAK_KEY") or None
```
Comment (pr0xity) [inline tgno/settings/base.py:198]: is there a reason for this to be different from the once belove?

Comment (olemathias) [reply tgno/settings/base.py:198]: Well, yes and no. I need it to be `None` when the variable is set but empty.
But there is no reason we can't do the same for all others. I just think the `env()` is more elegant.

I guess I could have moved the "if empty or not set" check where we use it.

Comment (pr0xity) [reply tgno/settings/base.py:198]: So this `SOCIAL_AUTH_KEYCLOAK_SECRET = env("SOCIAL_AUTH_KEYCLOAK_SECRET", None)` will sett it to `""` if it's empty and `SOCIAL_AUTH_KEYCLOAK_KEY = os.getenv("SOCIAL_AUTH_KEYCLOAK_KEY") or None` will sett it to `None` when it's empty?

Comment (pr0xity) [reply tgno/settings/base.py:198]: >  I just think the `env()` is more elegant.

I also think `env()` is more elegant 😅 
That's why I'm asking 😄

**Comment by mfyll** [comment]
Very nice 🚀

**Review by pr0xity — APPROVED** [review]
image
LGTM

File: tgno/settings/base.py (hunk 1 above)
Comment (olemathias) [reply tgno/settings/base.py:198]: Yes. I use SOCIAL_AUTH_KEYCLOAK_KEY if we should enable keycloak login or not. Do not want to enable it if the user is just copy/pasting the example .env file.

**Pull request merged**